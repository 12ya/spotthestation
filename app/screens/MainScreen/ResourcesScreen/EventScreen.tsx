import { useNavigation, useRoute } from "@react-navigation/native"
import { observer } from "mobx-react-lite"
import React from "react"
import { ViewStyle, TextStyle, Pressable, View } from "react-native"
import { useSafeAreaInsets } from "react-native-safe-area-context"
import { Icon, Text } from "../../../components"
import { colors, typography } from "../../../theme"
import { WebView } from "react-native-webview"
import { StyleFn, useStyles } from "../../../utils/useStyles"

export interface EventScreenRouteProps {
  item: any
}

export const EventScreen = observer(function EventScreen() {
  const navigation = useNavigation()
  const {
    params: { item },
  } = useRoute<any>()

  const { $headerStyleOverride, $backButton, $webViewContainer, $backButtonText } =
    useStyles(styles)

  const topInset = useSafeAreaInsets().top

  return (
    <View style={[$headerStyleOverride, { paddingTop: topInset }]}>
      <Pressable
        accessible
        accessibilityLabel="Back button"
        accessibilityHint="Navigates to the previous screen"
        accessibilityRole="button"
        onPress={() => navigation.goBack()}
        style={$backButton}
      >
        <Icon icon="caretLeft" color={colors.palette.neutral250} size={24} />
        <Text tx="resources.header" style={$backButtonText} />
      </Pressable>
      <WebView source={{ uri: item }} containerStyle={$webViewContainer} />
    </View>
  )
})

const styles: StyleFn = ({ scale, fontSizes, lineHeights }) => {
  const $headerStyleOverride: TextStyle = {
    backgroundColor: colors.backgroundDark,
    flex: 1,
  }

  const $backButton: ViewStyle = {
    flexDirection: "row",
    alignItems: "center",
    zIndex: 999,
    backgroundColor: colors.backgroundDark,
    width: "100%",
    paddingBottom: scale(11),
  }

  const $webViewContainer: ViewStyle = {
    flex: 1,
  }

  const $text: TextStyle = {
    fontFamily: typography.primary?.normal,
    fontSize: fontSizes[18],
    lineHeight: lineHeights[24],
    color: colors.palette.neutral450,
    textAlign: "left",
    paddingBottom: scale(24),
  }

  const $backButtonText: TextStyle = {
    ...$text,
    color: colors.palette.neutral250,
    paddingBottom: 0,
    paddingLeft: scale(5),
  }

  return { $headerStyleOverride, $backButton, $webViewContainer, $text, $backButtonText }
}
