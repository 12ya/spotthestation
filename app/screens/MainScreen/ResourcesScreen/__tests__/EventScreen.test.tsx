import React from "react"
import { EventScreen } from "../EventScreen"
import { NavigationContainer } from "@react-navigation/native"
import { render } from "@testing-library/react-native"

jest.mock("@react-navigation/native", () => ({
  ...Object.assign({}, jest.requireActual("@react-navigation/native")),
  useRoute: () => ({
    params: {
      id: 1,
      item: {},
    },
  }),
}))
it("renders correctly", () => {
  const tree = render(
    <NavigationContainer>
      <EventScreen />
    </NavigationContainer>,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
