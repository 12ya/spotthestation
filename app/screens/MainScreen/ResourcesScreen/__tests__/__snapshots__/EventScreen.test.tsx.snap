// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0, 0, 0, 1)",
        "flex": 1,
      },
      Object {
        "paddingTop": 0,
      },
    ]
  }
>
  <View
    accessibilityHint="Navigates to the previous screen"
    accessibilityLabel="Back button"
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 1)",
        "flexDirection": "row",
        "paddingBottom": 16,
        "width": "100%",
        "zIndex": 999,
      }
    }
  >
    <View>
      <RCTImageView
        internal_analyticTag={null}
        resizeMode="contain"
        source={
          Array [
            Object {
              "height": undefined,
              "uri": undefined,
              "width": undefined,
            },
          ]
        }
        style={
          Object {
            "height": 35,
            "overflow": "hidden",
            "resizeMode": "contain",
            "tintColor": "#DEDDDE",
            "width": 35,
          }
        }
        tintColor="#DEDDDE"
      />
    </View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {},
          Array [
            Object {
              "fontSize": 23,
              "lineHeight": 35,
            },
            Object {
              "fontFamily": "Helvetica Neue",
            },
            Object {
              "color": "#191015",
            },
          ],
          undefined,
          undefined,
          Object {
            "color": "#DEDDDE",
            "fontFamily": "Helvetica Neue",
            "fontSize": 26,
            "lineHeight": 35,
            "paddingBottom": 0,
            "paddingLeft": 7,
            "textAlign": "left",
          },
        ]
      }
    >
      resources.header undefined
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "height": 100,
          "scale": 2,
          "uri": "https://placekitten.com/200/200",
          "width": 100,
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      textInteractionEnabled={true}
      useSharedProcessPool={true}
    />
  </View>
</View>
`;
